src/core/ext/transport/binder/wire_format/wire_reader_impl.h

```diff
@@ -99,9 +101,11 @@ class WireReaderImpl : public WireReader {
  private:
   absl::Status ProcessStreamingTransaction(transaction_code_t code,
                                            ReadableParcel* parcel);
-  absl::Status ProcessStreamingTransactionImpl(transaction_code_t code,
-                                               ReadableParcel* parcel,
-                                               int* cancellation_flags)
+  absl::Status ProcessStreamingTransactionImpl(
+      transaction_code_t code, ReadableParcel* parcel, int* cancellation_flags,
+      // The queue saves the actions needed to be done "WITHOUT" `mu_`.
+      // It prevents deadlock against wire writer issues.
+      std::queue<std::function<void()>>& deferred_func_queue)
```

Comment:
Have you considered inline this function and reconsider how to extract it in a logically way? Currently it requires caller to acquire `mu_` while calling it but it needs to do some tasks without holding`mu_`, which shows current way we structure the code probably doesn't make sense anymore.

Reply:
I think it's fine that we do somethings needing `mu_` in `ProcessStreamingTransactionImpl()` and do something without `mu_` in `ProcessStreamingTransaction()`?

Reply:
Yeah it is fine, just looks strange. The function declaration is currently annotated with `ABSL_EXCLUSIVE_LOCKS_REQUIRED(mu_)`, and we made it a separate function because we want to put codes that need to be executed while holding `mu_` in it. Now, if we just inline the function, we can simply use local variables and will not need all these callback stuff that is hard to read